internal/transport/http2_client.go

```diff
@@ -153,8 +154,12 @@ type http2Client struct {
 	logger       *grpclog.PrefixLogger
 }
 
-func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error), addr resolver.Address, useProxy bool, grpcUA string) (net.Conn, error) {
+func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error), addr resolver.Address, grpcUA string) (net.Conn, error) {
 	address := addr.Addr
+
+	if opts, present := proxyattributes.Get(addr); present {
+		return proxyDial(ctx, addr, grpcUA, opts)
+	}
```

Comment:
Why do we want this to take higher precedence than a custom dialer?

I see that we later make it so a custom dialer disables the delegating resolver that would set the proxy attribute.  But what about if xDS is used and the proxy is set that way, and the user has a custom dialer?  I would expect the custom dialer to have higher precedence, probably, since that one is set directly by the application.

It is admittedly a weird case, however.